src/Text/Pandoc/Writers/Typst.hs

```diff
@@ -285,7 +285,7 @@ blockToTypst block =
                 $$ footer
             )
             $$ ")"
-      return $ case Data.List.find (== "typst-no-figure") tabclasses of
+      return $ case Data.List.find (== "typst:no-figure") tabclasses of
```

Comment:
I think it's more straightforward to use a boolean test here:  `if "typst:no-figure" `elem` tabclasses`